test/unit/lib/GitHub.test.js

```diff
@@ -284,25 +316,23 @@ describe('GitHub interface', () => {
         }
       })
         .put('/repos/johndoe/foobar/contents/path/to/file.txt')
-        .reply(200, {
-          number: 123
-        })
+        .replyWithError('{ message: "an error" }')
```

Comment:
This test was failing after it was fixed to actually check if an exception was thrown. ISTM we are returning status 200 OK, so it isn't clear why we expect a failure. Changed to a failing reply and it passes.